File: app/src/main/java/id/coedotz/appbiodata/api/Retroserver.java
package id.coedotz.appbiodata.api;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class Retroserver {
    private  static  final String base_url = "https://proyek.luckytruedev.com/johancrud/";

    private static Retrofit retrofit;


    public static Retrofit getClient()
    {
        if(retrofit == null)
        {
            retrofit = new Retrofit.Builder()
                    .baseUrl(base_url)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }

        return  retrofit;
    }
}

File: app/src/main/java/id/coedotz/appbiodata/api/Retroserver.kt
package id.coedotz.appbiodata.api

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object Retroserver {
    private const val base_url = "https://proyek.luckytruedev.com/johancrud/"
    private var retrofit: Retrofit? = null

    fun getClient(): Retrofit {
        if (retrofit == null) {
            retrofit = Retrofit.Builder()
                .baseUrl(base_url)
                .addConverterFactory(GsonConverterFactory.create())
                .build()
        }

        return retrofit!!
    }
}
